.github/workflows/binding-checks.yml

```diff
@@ -2,46 +2,57 @@ name: Binding Checks
 
 on:
   pull_request:
-    paths:
-      - # Any new or changed canisters
-        'dfx.json'
-      - # Any new or changed .did files
-        '**/*.did'
-      - # Scripts, GitHub actions that contain 'backend' in their path.
-        '**/*backend*'
-      - # The backend source code
-        'src/backend/**'
-      - 'src/shared/**'
-      - # Rust files such as Cargo.lock, Cargo.toml and rust-toolchain.toml
-        '**/Cargo*'
-      - '**/*rust*'
-      - # This workflow
-        '.github/workflows/binding-checks.yml'
   workflow_dispatch:
-
 jobs:
-
   generate:
     runs-on: ubuntu-22.04
     steps:
       - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v3
+        id: changes
+        with:
+          filters: |
+            src:
+              - # Any new or changed canisters
+                'dfx.json'
+              - # Any new or changed .did files
+                '**/*.did'
+              - # Scripts, GitHub actions that contain 'backend' in their path.
+                '**/*backend*'
+              - # The backend source code
+                'src/backend/**'
+              - 'src/shared/**'
+              - # Rust files such as Cargo.lock, Cargo.toml and rust-toolchain.toml
+                '**/Cargo*'
+              - '**/*rust*'
+              - # Scripts that download external Candid files, with false positives.
+                'scripts/build.*.sh'
+              - # This workflow
+                '.github/workflows/binding-checks.yml'
```

Comment:
Note: Path filters moved here so that we can make this workflow required for merge, but not have workflows hang when a relevant file hasn't changed.